src/NUnitFramework/framework/Diagnostics/ProgressTraceListener.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Diagnostics;
+#if !NET6_0_OR_GREATER
+using System.Security.Permissions;
+#endif
+
+namespace NUnit.Framework.Diagnostics
+{
+    /// <summary><para>
+    /// The ProgressTraceListener class allows directing tracing or
+    /// debugging output to <see cref="TestContext.Progress"/>.
+    /// </para><para>
+    /// To activate, place the following snippet into the one-time
+    /// set-up method of either a test's fixture or the set-up
+    /// fixture of a project:
+    /// <c>
+    /// System.Trace.Listeners.Add(new ProgressTraceListener());
+    /// </c>
+    /// </para><para>
+    /// Make sure to only add a listener once, e.g.:
+    /// <c>
+    /// if (!System.Trace.Listeners.OfType&lt;ProgressTraceListener&gt;().Any())
+    ///     System.Trace.Listeners.Add(new ProgressTraceListener());
+    /// </c>
+    /// </para><para>
+    /// Alternatively, add it in the one-time set-up and again remove
+    /// it in the one-time tear-down, e.g.:
+    /// <c>
+    /// _progressTraceListener = new ProgressTraceListener();
+    /// System.Trace.Listeners.Add(_progressTraceListener);
+    /// </c>
+    /// <c>
+    /// System.Trace.Listeners.Remove(_progressTraceListener);
+    /// _progressTraceListener.Close();
+    /// </c>
+    /// </para></summary>
+    /// <remarks><para>
+    /// Although named "Trace", <see cref="TextWriterTraceListener"/>
+    /// "directs tracing or debugging output".
+    /// </para><para>
+    /// This listener is provided by NUnit (i.e. the origin of
+    /// <see cref="TestContext.Progress"/>) same as the
+    /// <see cref="ConsoleTraceListener"/> is provided by .NET
+    /// (the origin of <see cref="Console"/>).
+    /// </para></remarks>
+    // Implemented following the ConsoleTraceListener implementation.
+#if !NET6_0_OR_GREATER
+    [HostProtection(Synchronization = true)]
```

Comment:
I think I'm unclear if you needed this for your specific hosting requirements but I think I would prefer we remove this unless you explicitly need this @maettu-this , at which point we should have a discussion before adding it. 

The attribute, and code access security in general are deprecated, and the documentation mentions 
> This attribute affects only unmanaged applications that host the common language runtime and implement host protection, such as SQL Server. If the code is run in a client application or on a server that is not host-protected, the attribute "evaporates"; it is not detected and therefore not applied. 

Though even the docs for hosting CLR in SQL Server also mention it is no longer supported there
https://learn.microsoft.com/en-us/sql/relational-databases/clr-integration/common-language-runtime-integration-overview?view=sql-server-ver16#code-access-security-no-longer-supported

Deprecated functionality has a way of being "sticky" and being hard to remove which may be why we still see it in reference source for the full framework. It was indeed sticky when we added CAS / partial trust support in NUnit as it took us 8 years and a major version before it could be removed.

Reply:
I'm fine with removing it, I'll push the removal in a minute.